### ISSUE TITLE
Fix PMD violation  JUnit5TestShouldBePackagePrivate

### ISSUE THREAD

**Issue by pbludov** [checkstyle/checkstyle#10063]
```
[INFO] PMD Failure: com.puppycrawl.tools.checkstyle.xpath.XpathQueryGeneratorTest:559 Rule:JUnit5TestShouldBePackagePrivate Priority:3 JUnit 5 tests should be package-private..
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-pmd-plugin:3.14.0:check (default) on project checkstyle: You have 4488 PMD violations.
```

New rule [JUnit5TestShouldBePackagePrivate](https://pmd.github.io/pmd-6.35.0/pmd_rules_java_bestpractices.html#junit5testshouldbepackageprivate) requires that all tests should be like
```java
    @Test
    /* package */ void testGood() {
    }
```

Another possibility is to remove just `public` modifier and disable [CommentDefaultAccessModifier](https://pmd.github.io/latest/pmd_rules_java_codestyle.html#commentdefaultaccessmodifier) for tests.

_Originally posted by @pbludov in https://github.com/checkstyle/checkstyle/issues/10055#issuecomment-851742376_

**Comment by romani** [comment]
I do not like package and protected visibility, they are HACK in design and illusion of protection.
if nothing to protect - make it public.
We have the same violation in Sonar, and I do not think code will benefit from such rule.

**Comment by pbludov** [comment]
I prefer public test methods too.  I close this issue, if someone wants to discuss it, please reopen it.
